src/js/components/AppListComponent.jsx

```diff
@@ -186,7 +198,11 @@ var AppListComponent = React.createClass({
 
     if (filterText != null && filterText !== "") {
       nodesSequence = nodesSequence
-        .filter(app => app.id.indexOf(filterText) !== -1);
+        .filter(app => {
+          app.filterScore = score(app.id, filterText);
```

Comment:
You are modifying the ```app``` here, this should be done only in the store (with emiting an event afterwards).
Also this is done in a ```filter```. It's not expectable that a filter modifies data.
So this isn't the right approach. Sorry. :)

Let's try to sort and score the list where the sorting belongs to.
Could you add a sort function after the .sort()-function at line 316?
```javascript
      if (filterText != null) {
        nodesSequences.sort((a, b) => {
          return score(a.id, filterText) > score(b.id, filterText)
            ? -1
            : 1;
        });
      }
```
Didn't tested it, but it should go into this direction.
Maybe there is also a more efficient way by integrating this into the already existing sort-function. Not sure.

This doesn't modify the app itself. :)